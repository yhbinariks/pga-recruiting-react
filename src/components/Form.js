import React from 'react';
import PropTypes from 'prop-types';

const Form = ({
  player,
  action,
  onSubmit,
}) => (
  <form onSubmit={onSubmit} className="player-form">
    <label className="player-form__label">
      <span className="player-form__label-text">First Name</span>
      <input defaultValue={player.firstName} name="firstName" type="text" required maxLength="255" />
    </label>
    <label className="player-form__label">
      <span className="player-form__label-text">Last Name</span>
      <input defaultValue={player.lastName} name="lastName" type="text" required maxLength="255" />
    </label>
    <label className="player-form__label">
      <span className="player-form__label-text">Score</span>
      <input defaultValue={player.score} name="score" type="number" min="0" max="100" required />
    </label>
    <button className="player-form__submit" type="submit">{action}</button>
  </form>
);

const defaultProps = {
  player: {
    firstName: '',
    lastName: '',
    score: 0,
  },
};

const propTypes = {
  player: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
  }),
  action: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;


export default Form;
