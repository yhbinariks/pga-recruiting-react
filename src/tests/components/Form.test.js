import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import Form from '../../components/Form';


describe('Form component', () => {
  it('snapshot defaultProps player', () => {
    const props = {
      action: 'add',
      onSubmit: jest.fn(),
    };
    const form = shallow(<Form {...props} />);
    expect(toJson(form)).toMatchSnapshot();
  });
  it('snapshot', () => {
    const props = {
      player: {
        firstName: 'petro',
        lastName: 'chu',
        score: 32,
      },
      action: 'add',
      onSubmit: jest.fn(),
    };
    const form = shallow(<Form {...props} />);
    expect(toJson(form)).toMatchSnapshot();
  });

  it('submit', () => {
    const props = {
      action: 'add',
      onSubmit: jest.fn(),
      player: {
        firstName: 'petro',
        lastName: 'chu',
        score: 32,
      },
    };
    const form = mount(<Form {...props} />);
    form.find('form').simulate('submit');
    expect(props.onSubmit).toHaveBeenCalledTimes(1);
  });
});
