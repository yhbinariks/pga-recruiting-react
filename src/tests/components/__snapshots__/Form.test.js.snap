// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot 1`] = `
<form
  className="player-form"
  onSubmit={[MockFunction]}
>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      First Name
    </span>
    <input
      defaultValue="petro"
      maxLength="255"
      name="firstName"
      required={true}
      type="text"
    />
  </label>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      Last Name
    </span>
    <input
      defaultValue="chu"
      maxLength="255"
      name="lastName"
      required={true}
      type="text"
    />
  </label>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      Score
    </span>
    <input
      defaultValue={32}
      max="100"
      min="0"
      name="score"
      required={true}
      type="number"
    />
  </label>
  <button
    className="player-form__submit"
    type="submit"
  >
    add
  </button>
</form>
`;

exports[`Form component snapshot defaultProps player 1`] = `
<form
  className="player-form"
  onSubmit={[MockFunction]}
>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      First Name
    </span>
    <input
      defaultValue=""
      maxLength="255"
      name="firstName"
      required={true}
      type="text"
    />
  </label>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      Last Name
    </span>
    <input
      defaultValue=""
      maxLength="255"
      name="lastName"
      required={true}
      type="text"
    />
  </label>
  <label
    className="player-form__label"
  >
    <span
      className="player-form__label-text"
    >
      Score
    </span>
    <input
      defaultValue={0}
      max="100"
      min="0"
      name="score"
      required={true}
      type="number"
    />
  </label>
  <button
    className="player-form__submit"
    type="submit"
  >
    add
  </button>
</form>
`;
